crates/trie/sparse/src/trie.rs

```diff
@@ -571,8 +571,25 @@ impl RevealedSparseTrie {
     pub fn update_rlp_node_level(&mut self, depth: usize) {
         let targets = self.get_nodes_at_depth(depth);
         let mut prefix_set = self.prefix_set.clone().freeze();
+
+        // stack of paths we need rlp nodes for
+        let mut path_stack = Vec::new();
+        // stack of rlp nodes
+        let mut rlp_node_stack = Vec::<(Nibbles, RlpNode)>::new();
+        // reusable branch child path
+        let mut branch_child_buf = SmallVec::<[Nibbles; 16]>::new_const();
+        // reusable branch value stack
+        let mut branch_value_stack_buf = SmallVec::<[RlpNode; 16]>::new_const();
```

Comment:
worth creating a helper struct for this mb? `RlpNodeBufs` or smth

Reply:
agreed, added a struct